oval_graph/oval_node.py

```diff
@@ -16,60 +16,69 @@ class OvalNode():
     '''
     The OvalNode object is one node of oval_graph.
 
-    Args:
+    Require args:
         node_id (str|int): identifies node
-        input_node_type (str): type of node (value or operator)
-        input_value (str): value of node
-        input_negation (bool): value indicating whether the node is negated or not
-        comment (str): text about node
-        tag (str): tag specifies if the node represents OVAL test, OVAL definition or XCCDF rule
-        data (dict|None): information about test
-        children ([OvalNode]): array of children of node
+        node_type (str): type of node (value or operator)
+        value (str): value of node
+
+    Optional args:
+        negation (bool): value indicating whether the node is negated (empty eq False)
+        comment (str): text about node (empty eq None)
+        tag (str): tag specifies if the node represents OVAL test,
+        OVAL definition or XCCDF rule (empty eq None)
+        test_result_details (dict|None): information about test (empty eq None)
+        children ([OvalNode]): array of children of node (empty eq empty array)
 
     Attributes:
         node_id (str): id of node
         node_type (str): type node
-        value (str): value of node for operator and,
-        or, one etc... and for value true, false, error etc...
+        value (str): value of node for operator and, or, one etc... and for value true,
+        false, error etc...
+        negation (bool): value indicating whether the node is negated
+        comment (str): some comment about node
+        tag (str): tag specifies if the node represents OVAL test,
+        OVAL definition or XCCDF rule
+        test_result_details (dict|None): information about test
         children ([OvalNode]): children of node
     '''
 
-    def __init__(
-            self,
-            node_id,
-            input_node_type,
-            input_value,
-            input_negation,
-            comment,
-            tag,
-            test_result_details,
-            children=None
-    ):
-        self.comment = comment
-        self.node_id = node_id
-        self.negation = self.validate_negation(input_negation)
-        self.node_type = self.validate_type(input_node_type)
-        self.value = self.validate_type_and_value(input_value)
+    def __init__(self, **kwargs):
+        try:
+            assert 'node_id' in kwargs
+            assert 'node_type' in kwargs
+            assert 'value' in kwargs
+            self.node_id = kwargs.get('node_id')
+            self.node_type = self._validate_type(kwargs.get('node_type'))
+            self.value = self._validate_value(kwargs.get('value'))
+
+            self.negation = self._validate_negation(
+                kwargs.get('negation', False))
+        except AssertionError:
```

Comment:
asserts may be dropped from the code if it is compiled with the `-O` flag. So you can write more general `except Exception as exc`, and then include the `str(exc)` message as an indication of what went wrong.